### PR TITLE
Added mix mode button to Shade

### PR DESCRIPTION
**Pull request by daschuer**
bildschirmfoto vom 2018-06-04 00 38 14
left "Add Wet to Dry" right "Dry Wet Crossover"

### CODE REVIEW

File: res/skins/Shade/deck_effect.xml [hunk 1]
```diff
@@ -97,33 +97,53 @@
...
 				<Path>knobs/knob_rotary_s%1.png</Path>
-				<Pos>49,29</Pos>
+				<Pos>49,24</Pos>
 				<Connection>
 					<ConfigKey>[EffectRack1_EffectUnit<Variable name="effectunitnum"/>],mix</ConfigKey>
 				</Connection>
 			</Knob>
 
+		    <PushButton>
+			    <TooltipId></TooltipId>
```
Comment (Be-ing) [inline res/skins/Shade/deck_effect.xml]: EffectUnit_mix_mode

**Review by Be-ing — CHANGES_REQUESTED** [review]
Interesting idea with the graphics. However, the crossing lines for D/W mode look like the letter "X" to me, which suggests that something is being blocked or disabled.

**Comment by daschuer** [comment]
OK, tooltips are in place. 
I have played a bit with the X, but everything melds into noise when lowering the size. 
I am actually pleased with the current bitmap. Even when someone thinks there is something disables, he may click on it and understands the hopefully the intention. Dry signal is blocked ...

**Comment by Be-ing** [comment]
This branch now has unrelated commits. Please rebase it on master.

**Comment by Be-ing** [comment]
> when someone thinks there is something disables, he may click on it and understands the hopefully the intention. Dry signal is blocked ...

I don't understand what you mean by blocking the dry signal. Could you explain?

**Comment by daschuer** [comment]
Nothing important. In case you turn fully right, dry signal is blocked. 
Rebased.

**Comment by Be-ing** [comment]
I have an idea for a graphic to represent D/W mode: a simple horizontal bar with gradient between different colors on each end. What do you think?

**Comment by daschuer** [comment]
How will the D+W mode look than? If we do a gradient the same way, it will be hard to see the difference.
D+W will just be darker on the right, because of more dry signal.

**Comment by ronso0** [comment]
I think the initial icons are okay, but not knowing what they are about I'd think they are descriptive graphics for the knob above but not buttons.
the crossing lines are 
would they still fit in there when they have a border, after moving the knobs & labels a few pixels up?

**Comment by ronso0** [comment]
I would introduce gradients in Shade. If you think the liens aren't clear enough flat triangles could work IMO

**Comment by daschuer** [comment]
Yes we can add a border. Currently the match with the invert and metknob linking buttons, which have also no border: I think both should look similar: 
bildschirmfoto vom 2018-06-07 22-54-02

**Comment by daschuer** [comment]
d w
bildschirmfoto vom 2018-06-07 22-54-02

**Comment by daschuer** [comment]
The border less version looks more elegant though.

**Comment by Be-ing** [comment]
> How will the D+W mode look than? If we do a gradient the same way, it will be hard to see the difference.
D+W will just be darker on the right, because of more dry signal.

Almost the same as the current D+W icon, but I think it would help if the horizontal bar representing the dry signal was the same height as the maximum height of the triangle representing the wet signal.

**Comment by Be-ing** [comment]
@ronso0 I think a gradient is a good symbolic representation of crossfading between two signals.

**Comment by Be-ing** [comment]
> flat triangles could work

A symbol like `><` implies there is nothing in the middle. A symbol like `<>` implies there is nothing at the sides. What may work would be a long rectangle with the diagonal dividing two colors on top and bottom.

**Comment by daschuer** [comment]
It e put the dry signal on the top, both graphics follow the view of the crossfader curve in preferences. 
That should work IMHO.

**Comment by Be-ing** [comment]
What do you think of my sketch for the D/W icon?

**Comment by daschuer** [comment]
image
image

**Comment by daschuer** [comment]
> What do you think of my sketch for the D/W icon?

The D+W icon does not work that way.

**Comment by Be-ing** [comment]
Right, I was suggesting that sketch for the D/W icon, not D+W.

**Comment by Be-ing** [comment]
No, the crossed lines do not convey what the D/W mode does.

**Comment by daschuer** [comment]
On the right you have more wet and less dry. In the middle you have the same level for both. It works for me in the same way the fader curve visualization works. So it is at least now wrong. If you switch the buttons, you van now also see that the dry signal is increased on the right end. Exactly that happens with the sound.

**Comment by Be-ing** [comment]
I think the crossed lines for D/W look too much like an icon indicating that something is disabled or an error to convey what it does.

**Comment by daschuer** [comment]
I think it is finally a matter of taste.
It works for me quite good and as I understand it also for Ronso0. We have no better alternative yet. Feel free to propose alternatives for the other skins. 

If space is sufficient, the D/W|D+W seems to ba a good alternative.

**Comment by Be-ing** [comment]
You still have not responded to my [proposal for D/W](https://github.com/mixxxdj/mixxx/pull/1702#issuecomment-395570216).

**Comment by daschuer** [comment]
How will D+W look compatef to you proposal? We need to consider both at once.

**Comment by Be-ing** [comment]
You have already asked and I [already answered](https://github.com/mixxxdj/mixxx/pull/1702#issuecomment-395567340).

**Comment by daschuer** [comment]
This was you answer to the gradent D/W button. This old idea does IMHO not match your new design.

**Comment by Be-ing** [comment]
> This old idea does IMHO not match your new design.

I don't know what you mean. What doesn't match what?

I have realized there is a UX problem with only using an icon to identify the states of this switch. How can we teach users the names of the different states? If we refer to "D/W" and "D+W" in the tooltip but the icons are just abstract graphical representations of the concepts, users won't know what the tooltip is talking about. Any ideas to overcome this?

**Comment by daschuer** [comment]
> I don't know what you mean. What doesn't match what?

You have not yet proposed two matching buttons. My latest proposal can be read the same way for both buttons and has a nice animation when changing the state. 

I have no issue with this: X id probably the best association for "Crossfade" that we can use. See:
image D/W mode: Mix knob crossfades between dry and wet

image D+W mode: Mix knob adds wet to dry. Use this to change only the effected (wet) signal with EQ and filter effects.

**Comment by Be-ing** [comment]
I described my [idea for the D+W icon](https://github.com/mixxxdj/mixxx/pull/1702#issuecomment-395567340) above. The original icon in the first commit was better than the current one IMO. The current one should at least be reflected horizontally so the wider side of the triangle is on the right. I think it looks a little odd to put the triangle below the straight line, but whether it's above or below doesn't really change the meaning.

I don't think I've ever seen an "X" icon in a GUI mean anything other than "error" or "close". I think it's a good symbol for crossfading.

**Comment by daschuer** [comment]
> I described my idea for the D+W icon above.

I have already stated that I think this old idea does not match your new design. Please propose a mock-up for both buttons.

**Comment by ronso0** [comment]
Again, I think the both icons do work as intended.
The only remark I have: because the icon is so tiny there's an illusion that makes the flat X look a bit like this (exaggerated)
ex_1
Try to add a short vertical section in the source svg where those two lines cross to make it look as intended (a bit like those tricks used when creating fonts/letters):
ex_2

**Comment by ronso0** [comment]
I think I'll use similiar icons for Tango. They are more compact than text labels.

**Comment by Be-ing** [comment]
Please do not use anything that looks like an X to signify something that is not an error or problem.

**Comment by Be-ing** [comment]
Ping. Thoughts about the above proposal?

**Comment by ronso0** [comment]
>  Thoughts about the above proposal?

The space below the Mix knob is like 7x24 px. I don't see how icons you proposed should fit..

**Comment by daschuer** [comment]
bildschirmfoto vom 2018-07-06 16 05 20

ready

**Comment by Be-ing** [comment]
This still looks like an "X"...

**Comment by daschuer** [comment]
But you see the waveform, right? We simply have no better proposal jet, so can't we just merge it and wait if we get real user complains about the X.

**Comment by Be-ing** [comment]
I don't understand what the waveform has to do with this. These icons are meaningless in the context of the tooltip too. No one will look at these icons and know that they correspond to states called "D+W" and "D/W". Can we simply use the "D+W" and "D/W" text for now?

**Comment by Be-ing** [comment]
Can't you put a border around the button to show it is a button instead of just a text label?

**Comment by daschuer** [comment]
I have actually tried first to use an original shade button. But it is too big. These bars are already established as buttons. So we can just reuse the routing linking from knobs. But the current solution works for me much better. I can see here the mixing behaviour. By the way much better that only see a + vs / but it is a matter of taste. (like the whole skin by the way)

**Comment by daschuer** [comment]
Done! Thank you for the buttons.

**Comment by Be-ing** [comment]
Following discussion in #1748 to adopt these designs and change the mix mode tooltip, I think this is okay to merge. The mix mode icons somehow still look blurry even though they are SVGs, but that's also the case for the whole skin so I don't think that should hold up merging this PR.

**Pull request merged**